File: asticaVoice/node/asticaVoice_voice_clone_delete.js
#!/usr/bin/env node
/**
 * Delete a user-specific voice clone via astica Voice API (Node.js).
 *
 * - Calls POST https://voice.astica.ai/api/voice_clones/:id/delete
 * - Body: { "tkn": "<API_KEY>" }
 *
 * NOTE:
 *   The ":id" here refers to the internal clone row ID (not clone_id).
 *   You can obtain it from your own backend or other admin interfaces.
 *
 * Saves response to: voice-clone-delete.json
 *
 * Requirements:
 *   - Node.js 18+
 *
 * Usage:
 *   1) Set CLONE_ROW_ID below.
 *   2) node voice_clone_delete.js
 */

"use strict";

const fs = require("fs");
const path = require("path");

const ASTICA_API_KEY = "YOUR_API_KEY_HERE"; // obtain free api key -> https://astica.ai/api-keys/
const ASTICA_API_ENDPOINT = "https://voice.astica.ai";

// Internal DB row ID to delete (not clone_id):
const CLONE_ROW_ID = 1;

const OUTPUT_JSON_PATH = path.resolve("voice-clone-delete.json");

async function main() {
  if (!ASTICA_API_KEY || ASTICA_API_KEY === "YOUR_API_KEY_HERE") {
    console.error("Please set ASTICA_API_KEY at the top of this file.");
    process.exitCode = 1;
    return;
  }

  if (!CLONE_ROW_ID) {
    console.error("Please set CLONE_ROW_ID to the numeric clone row ID you want to delete.");
    process.exitCode = 1;
    return;
  }

  const url = `${ASTICA_API_ENDPOINT}/api/voice_clones/${CLONE_ROW_ID}/delete`;
  const payload = { tkn: ASTICA_API_KEY };

  console.log(`Deleting clone row ID=${CLONE_ROW_ID} via ${url} ...`);

  let resp;
  try {
    resp = await fetch(url, {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify(payload)
    });
  } catch (e) {
    console.error("[Network/HTTP error]", e.message || e);
    process.exitCode = 1;
    return;
  }

  let bodyText;
  let body;
  try {
    bodyText = await resp.text();
    body = JSON.parse(bodyText);
  } catch {
    body = bodyText || "";
  }

  if (!resp.ok) {
    console.error(`HTTP ${resp.status}`);
    console.error(typeof body === "string" ? body : JSON.stringify(body, null, 2));
    process.exitCode = 1;
    return;
  }

  console.log("Delete response:");
  console.log(JSON.stringify(body, null, 2));

  fs.writeFileSync(OUTPUT_JSON_PATH, JSON.stringify(body, null, 2), "utf8");
  console.log();
  console.log(`Response saved to: ${OUTPUT_JSON_PATH}`);
}

if (require.main === module) {
  main().catch((err) => {
    console.error("Unexpected error:", err);
    process.exitCode = 1;
  });
}

File: asticaVoice/python/asticaVoice_voice_clone_delete.py
#!/usr/bin/env python3
"""
Delete a user-specific voice clone via astica Voice API.

- Calls POST https://voice.astica.ai/api/voice_clones/:id/delete
- Body: { "tkn": "<API_KEY>" }

IMPORTANT:
  The ":id" here refers to the internal clone row ID (not clone_id).
  Depending on your deployment, you may expose this ID in your clone list
  or manage deletions from the astica dashboard.

Saves response JSON to voice-clone-delete.json.

Setup:
    pip install -r requirements.txt

Usage:
    1) Set CLONE_ROW_ID to the numeric clone row ID you want to delete.
    2) python voice_clone_delete.py
"""

import json
from pathlib import Path

import requests


ASTICA_API_KEY = "YOUR_API_KEY_HERE"  # obtain free api key -> https://astica.ai/api-keys/
ASTICA_API_ENDPOINT = "https://voice.astica.ai"

# NOTE: This is the internal DB row ID, not the "clone_id" returned from list.
CLONE_ROW_ID = 1

OUTPUT_JSON_PATH = Path("voice-clone-delete.json")



def main() -> int:
    if not ASTICA_API_KEY or ASTICA_API_KEY == "YOUR_API_KEY_HERE":
        print("Please set ASTICA_API_KEY at the top of this file.")
        return 1

    if not CLONE_ROW_ID:
        print("Please set CLONE_ROW_ID to the numeric clone row ID you want to delete.")
        return 1

    url = f"{ASTICA_API_ENDPOINT}/api/voice_clones/{CLONE_ROW_ID}/delete"
    payload = {"tkn": ASTICA_API_KEY}

    print(f"Deleting clone row ID={CLONE_ROW_ID} via {url} ...")

    try:
        resp = requests.post(url, json=payload, timeout=30)
    except requests.RequestException as e:
        print(f"[Network/HTTP error] {e}")
        return 1

    try:
        body = resp.json()
    except Exception:
        body = {"raw": resp.text}

    if not resp.ok:
        print(f"HTTP {resp.status_code}")
        print(json.dumps(body, indent=2))
        return 1

    print("Delete response:")
    print(json.dumps(body, indent=2))

    OUTPUT_JSON_PATH.write_text(json.dumps(body, indent=2), encoding="utf-8")
    print()
    print(f"Response saved to: {OUTPUT_JSON_PATH.resolve()}")

    return 0


if __name__ == "__main__":
    raise SystemExit(main())
